### ISSUE TITLE
Support ScalaPB's package/file scoped options

### ISSUE THREAD

**Issue by adarshaj** [akka/akka-grpc#987]
### Short description

It is desirable to configure [file/package scoped scalapb options](https://scalapb.github.io/customizations.html#file-level-options) in proto file.

### Details

Currently akka-grpc codegen seems to use many of scalapb's options, however, there's a parity introduced by not supporting full set of options as supported by scalapb. This leads to a problem, where the files generated by scalapb would be following the specs as defined in their tool and akka-grpc would ignore those specs and generate a buggy code (for parts not handled by scalapb).

Here's a very simple example:
`package.proto` with following content:
```protobuf
syntax = "proto3";

package rpc;

import "scalapb/scalapb.proto";

option (scalapb.options) = {
  scope: PACKAGE
  package_name: "com.example.rpc"
  no_default_values_in_constructor: true
  lenses: false
};
```

`foo_service.proto` file with following definition:
```protobuf
syntax = "proto3";

package rpc;

service FooService {
  rpc GetSomething(FooRequest) returns (FooResponse);
}
message FooRequest {
}
message FooResponse {
}
```

Now the files are split across two different packages:
rpc/FooService{,Client}.scala : These two files use `package rpc` and generate types that are all prefixed with `rpc`, viz., `rpc.FooRequest` etc.

and another package (written by scalapb):
com/example/rpc/{FooServiceProto,FooRequest,FooResponse}.scala and generate types that are all prefixed with `com.example.rpc`.

Because of two different packages, the compilation will fail.

The workaround would be obviously to use `java_package`, but its a pita in a large project, having a single package scoped option makes it simple to handle this in one place.

There has been earlier effort in resolving parity: #651, #90,  #82 ; however, since scalapb keeps changing, the parity keeps coming back. Can this be solved in a reliable way?

**Comment by raboof** [comment]
> Now after compiling the project to generate proto sources, the files are split across two different packages

I agree this is a problem and I think we should update akka-grpc to be able to interpret the scalapb package/file-scoped options, and even allow our own settings to be applied in the same way (#451).

> since scalapb keeps changing, the parity keeps coming back. Can this be solved in a reliable way?

I don't think it is possible to anticipate *all* possible changes in scalapb, but I think re-using code from the scalapb library (like in https://github.com/akka/akka-grpc/pull/90/files#diff-475941c1763330c0c8fdd87a2ab4a263R5) and supporting package/file-scoped option (rather than passing them on like in #651) would improve the situation a lot.
